pd-client/client.go

```diff
@@ -82,19 +82,54 @@ func (c *Client) GetTS() (int64, int64, error) {
 	return req.physical, req.logical, err
 }
 
-// GetRegion get a region from PD by key.
+// GetRegion gets a region from PD by key.
 // The region may expire after split. Caller is responsible for caching and
-// take care of region change.
+// taking care of region change.
 func (c *Client) GetRegion(key []byte) (*metapb.Region, error) {
-	req := &regionRequest{
-		key:  key,
+	req := &metaRequest{
+		pbReq: &pdpb.GetMetaRequest{
+			MetaType:  pdpb.MetaType_RegionType.Enum(),
+			RegionKey: key,
+		},
 		done: make(chan error),
 	}
 	c.workerMutex.RLock()
 	c.worker.requests <- req
 	c.workerMutex.RUnlock()
 	err := <-req.done
-	return req.region, err
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	region := req.pbResp.GetRegion()
+	if region == nil {
+		return nil, errors.Errorf("[pd] region field in rpc response not set")
```

Comment:
And also can we check return value from an error variable instead of by checking nil?